### DISCUSSION TITLE
Reset a form with a nested field array

### DISCUSSION THREAD

**Discussion by mcAnastasiou** [react-hook-form/react-hook-form#4483] [Q&A]
I try to reset a form with a nested array. I created a [sandbox](https://codesandbox.io/s/react-hook-form-reset-usefieldarray-forked-zp3ub?file=/src/index.js) for this

**Comment by deniercounter** [comment]
Seems you have @lines 64 and 73 the same key: 
`<li` `key={item.id}>`

Could it be, that the second @line 73 should be: `<li key={nestedItem.id}>`?

**Reply by licatajustin** [reply]
@deniercounter that would be needed, but doesn't seem to resolve the OP's problem, right?

**Reply by licatajustin** [reply]
@mcAnastasiou try this [CSB](https://codesandbox.io/s/adoring-cerf-wkrlw?file=/src/App.js) -- resetting the twice nested resources work as expected.

**Comment by SergeyWebDeveloper** [comment]
Here is a working option - https://codesandbox.io/s/react-hook-form-reset-usefieldarray-forked-rm7gs?file=/src/index.js
1) you need to specify nested name as - `names.0.nested`
2) in render nested field replace `item.id -> nestedItem.id`
